File: c/LeetCode_1637_WidestVerticalAreaBetweenTwoPointsContainingNoPoints.c
#include <stdio.h>
#include <stdlib.h>


int cmp(const void *a, const void *b) {
    return *(int*)a - *(int*)b;
}


int maxWidthOfVerticalArea(int** points, int pointsSize, int* pointsColSize) {
    int *xs = (int*) malloc(pointsSize * sizeof(int));
    for (int i = 0; i < pointsSize; ++i) {
        xs[i] = points[i][0];
    } 

    qsort(xs, pointsSize, sizeof(int), cmp);

    int max = 0;
    for (int i = 0; i < pointsSize - 1; ++i) {
        int tmp = points[i + 1] - points[i];
        if (tmp > max) {
            max = tmp;
        }
    }

    return max;
}


int main() {

    return 0;
}

File: python/LeetCode_1637_WidestVerticalAreaBetweenTwoPointsContainingNoPoints.py
def maxWidthOfVerticalArea(points) -> int:
    arr_x = []
    for n in points:
        arr_x.append(n[0])

    arr_x = sorted(arr_x)
    diff = 0
    for i in range(len(arr_x) - 1):
        tmp = arr_x[i + 1] - arr_x[i]
        if (tmp > diff):
            diff = tmp 

    return diff